docs/docs/integrations/providers/notdiamond.mdx

```diff
@@ -0,0 +1,73 @@
+# Not Diamond
+
+[Not Diamond](https://notdiamond.ai) offers an AI model router that automatically determines which LLM is best suited to respond to any query,
+improving LLM output quality by combining multiple LLMs into a **meta-model** that learns when to call each LLM.
+
+Not Diamond supports cost and latency tradeoffs, customized router training, and real-time router personalization. Learn more via [the documentation](https://notdiamond.readme.io/).
+
+## Installation
+
+```shell
+pip install langchain notdiamond
+```
+
+## Configuration
+
+- API keys: You need API keys from [Not Diamond](https://app.notdiamond.ai/keys) and any LLM you want to use.
+
+## Quick Start
+
+### Router recommendations with `NotDiamondRunnable`
+
+First, Not Diamond offers `NotDiamondRunnable`, which recommends a model for the given prompt or chain.
+
+```python
+from langchain_community.utilities.notdiamond import NotDiamondRoutedRunnable
+from notdiamond import NotDiamond, LLMConfig
+
+nd_client = NotDiamond(
+    llm_configs=[LLMConfig("openai", "gpt-4o"), LLMConfig("openai", "gpt-4o-mini")],
+    api_key=os.getenv("NOTDIAMOND_API_KEY"),
+    default=LLMConfig("openai", "gpt-4o"),
+)
+prompt = "Hello there! Please summarize this webpage. https://www.poetryfoundation.org/poems/46565/ozymandias"
+
+nd_runnable = NotDiamondRunnable(nd_client=nd_client)
+print(nd_runnable.model_select(prompt))
+```
+
+```python
+"openai/gpt-4o"
+```
```

Comment:
Note that gpt-4o cannot access websites. It will answer this correctly because it infers the poem from `/ozymandias` in the url, but if you ask it to summarize "https://www.poetryfoundation.org/poems/46565", it will hallucinate the response instead of correctly saying it's a 404. If we want to use this example, it might be more compelling to include Perplexity as an option. I think gpt-4o + perplexity is a much better way to showcase routing overall as they have pretty complementary strengths and weaknesses. On the other hand, I think the gpt-4o + gpt-4o-mini combination is a good way to showcase cost or latency tradeoffs (speaking of which, is that implemented in this integration?).

Reply:
Unfortunately this approach does not 

Cost and latency tradeoffs are supported via `NotDiamond` client configuration, described now at the bottom of the docs. I've added an example of `tradeoff` to this section.

Reply:
For what it's worth, adding support for Perplexity to [`init_chat_model()`](https://github.com/langchain-ai/langchain/blob/b5d670498f357f4bfdca5c420a950611338fda54/libs/langchain/langchain/chat_models/base.py#L300) looks fairly straightforward, and we can do it after merging our integration too.